### DISCUSSION TITLE
oAuth2 node sdk (quickbooks)

### DISCUSSION THREAD

**Discussion by bidpivot** [wasp-lang/wasp#1932] [Archived]
I tried to ask this question in the discord channel but I don't have an account and Discord did not let me create one.

Is it possible to enable authentication in my wasp app with OAuth2 / OpenID connect using the QuickBooks node sdk?  I would like to allow quickbooks users to authenticate in my app using their quickbooks credentials.  Is that possible at this moment with wasp or probably not a good idea?

**Comment by Martinsos** [comment]
That is not possible at the moment, although we do plan to allow adding any oauth2 provider yourself, next to the ones we support directly! But at the moment you will have to choose one of the existing ones.
@infomiho any additional thoughts?

**Reply by infomiho** [reply]
@bidpivot One thing that _might work_ and I haven't tested this ... but, if you use the Keycloak provider (and setup a Keycloak server of your own) maybe you can integrate Quickbooks with Keycloak and then Wasp's Keycloak integration to achieve what you want. If that works, then you'll have 1st party Wasp support.

You can always go down the route of custom API endpoints (https://wasp-lang.dev/docs/advanced/apis) which would probably be a challenging thing to setup and you'd lose all the benefits of Wasp's auth and Auth UI as well.

**Comment by bidpivot** [comment]
@infomiho @Martinsos  -- thank you very much for the response.  I will checkout the Keycloak idea but likely I will wait until wasp allows adding a 3rd party oauth provider for this particular app.  Nevertheless, I am very excited about this framework and excited to use it on  projects going forward.  Thanks again!

**Reply by Martinsos** [reply]
Awesome, we are here for further help :)!
